cscs-checks/prgenv/affinity_check.py

```diff
@@ -0,0 +1,155 @@
+                'ntasks-per-core': 1,
+                'multithreading': None,
+                'OMP_PLACES': 'cores',
+            },
+        }
+        self.variant = variant
+
+    def setup(self, partition, environ, **job_opts):
+        if partition.name == 'gpu':
+            self.num_cpus_per_task = (
+                self.cases[self.variant]['num_cpus_per_task'])
+        else:
+            self.num_cpus_per_task = (
+                self.cases[self.variant]['num_cpus_per_task'] * 3)
+
+        self.num_tasks = 1
+        self.variables  = {
+            'OMP_NUM_THREADS': str(self.num_cpus_per_task),
+            'OMP_PLACES': self.cases[self.variant]['OMP_PLACES']
+            # OMP_PROC_BIND is set to TRUE if OMP_PLACES is defined
+            # OMP_PROC_BIND values MASTER, CLOSE and SPREAD give the same
+            # result as OMP_PROC_BIND=TRUE
+        }
+        super().setup(partition, environ, **job_opts)
+        if self.cases[self.variant]['ntasks-per-core']:
+            self.job.options += ['--ntasks-per-core=%s' %
+                                 self.cases[self.variant]['ntasks-per-core']]
+
+
+@rfm.parameterized_test(['alternate_socket_filling'],
+                        ['consecutive_socket_filling'],
+                        ['single_task_per_socket_omp'])
+class SocketDistributionTest(AffinityTestBase):
+    def __init__(self, variant):
+        super().__init__(variant)
+        self.descr = 'Checking distribution of ranks and threads over sockets.'
+        self.valid_systems = ['daint:mc', 'dom:mc']
+        self.cases = {
+            'alternate_socket_filling': {
+                'ref_mc': 'alternate_socket_filling.txt',
+                'num_tasks': 36,
+                'num_cpus_per_task': 1,
+                'num_tasks_per_socket': 18,
+                'multithreading': False,
+                'cpu-bind': None,
+            },
+            'consecutive_socket_filling': {
+                'ref_mc': 'consecutive_socket_filling.txt',
+                'num_tasks': 36,
+                'num_cpus_per_task': 1,
+                'num_tasks_per_socket': None,
+                'multithreading': False,
+                'cpu-bind': 'rank',
+            },
+            'single_task_per_socket_omp': {
+                'ref_mc': 'single_task_per_socket_omp.txt',
+                'num_tasks': 2,
+                'num_cpus_per_task': 18,
+                'num_tasks_per_socket': 1,
+                'multithreading': False,
+                'cpu-bind': None,
+            },
+        }
+        self.num_tasks = self.cases[variant]['num_tasks']
+        self.num_cpus_per_task = self.cases[variant]['num_cpus_per_task']
+        self.num_tasks_per_socket = self.cases[variant]['num_tasks_per_socket']
+
+    def setup(self, partition, environ, **job_opts):
+        super().setup(partition, environ, **job_opts)
+        if self.cases[self.variant]['cpu-bind']:
+            self.job.launcher.options = ['--cpu-bind=%s' %
```

Comment:
Should we check if the scheduler is SLURM here? 

We should either clean the programming environments if the scheduler is not SLURM (skip the test) or fail immediately. @vkarak, what do you think?

Reply:
Is it checking that `self.job.launcher.registered_name == srun` enough to say that the scheduler is Slurm?

Reply:
@victorusu  Since currently the test is valid only for `daint/dom`, I think it's fine to do this assumption.